.github/workflows/ci.yml

```diff
@@ -10,36 +10,52 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        ghc:
-          # NB: PyF doesn't work with GHC <8.4 (guibou/PyF#94)
-          - "8.4.1"
-          - "8.6.1"
-          - "8.8.1"
-          - "8.10.1"
-          - "8.10.7" # there's an API breakage that only affects 8.10.7
-          - "9.0.1"
-          - "9.2.1"
-          - "9.2.2" # there's an API breakage introduced here
-    env:
-      CONFIG: "--enable-tests --enable-benchmarks"
     steps:
-      - uses: actions/checkout@v2
-      - uses: haskell/actions/setup@v1
-        id: setup-haskell-cabal
-        with:
-          ghc-version: ${{ matrix.ghc }}
-          cabal-version: "3.6.0.0"
-      - run: cabal v2-update
-      - run: cabal v2-freeze $CONFIG
-      - uses: actions/cache@v2
-        with:
-          path: |
-            ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
-            dist-newstyle
-          key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
-      - run: cabal v2-test all $CONFIG
+    - name: Install nix
+      uses: cachix/install-nix-action@v16
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+    - name: Checkout repositiory code
+      uses: actions/checkout@v3
+    - name: List files
+      run: |
+        ls ${{ github.workspace }}
+    - name: build for ghc 8.10.7
+      run: |
+        nix build --print-build-logs .#ghc8107_categorifier-category
+    - name: build for ghc 8.8.4
+      run: |
+        nix build --print-build-logs .#ghc884_categorifier-category
```

Comment:
Running the different versions here should really be in a matrix.

Reply:
Now using matrix. Thanks for the suggestion!